### ISSUE TITLE
Incorrect TS2790 error when removing property of the instance, that is defined in prototype

### ISSUE THREAD

**Issue by mnaoumov** [microsoft/TypeScript#56763]
### 🔎 Search Terms

TS2790

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/MYGwhgzhAECyCeBhcVoG8BQ1oDMCuAdsABQCU6W20ATgKYAue1B0ALAEwDclAvhnxmAB7AhHrQAlqPpgitaAF5oBWgHc4SFBDLdh0ydNnBa7RcrUbkkbaV0iIQkLQB0IIQHNiAIilijL-CIyLwAaAz85Z0CSUltBe0cXN08fQzl2KMIY0PCZdMyg2LtRRNcPbwB5ACMAK1pgemd3BgAFaiF6DvgAB1oKnGJfPONSAuyw6rqGptb2zvoevoGh-1HosjiMFcjoszJFAD5oAEZ2AGZihycylO3jMeCwu4CsjcvS5O9njPXSHO+HkV4iVrp8vJN6o1mvQ2h0ur1+oM0iNATkIdNobD5otEc81q8gRgAPRE6AASQIADcwCAJAATaC0ajtajQegQdgAdgAnAAGYmkgAqAAt5EJetRZAyhDhoGBoAByOm0Jz0WgK6DiplgTqsgC2eDE0CqYu69AkIhpzgwytV8meY04QA

### 💻 Code

```ts
class MyClass {
  func() {
    return 42;
  }
}

const instance = new MyClass();
const instance2 = new MyClass();
console.log("instance.func()", instance.func());
console.log("instance2.func()", instance2.func());
console.log("Object.getPrototypeOf(instance).func()", Object.getPrototypeOf(instance).func());

instance.func = () => 123;
console.log("instance.func()", instance.func());
console.log("instance2.func()", instance2.func());
console.log("Object.getPrototypeOf(instance).func()", Object.getPrototypeOf(instance).func());

// Invalid error ts2790
// The operand of a 'delete' operator must be optional.
delete instance.func;
```

### 🙁 Actual behavior

`Error ts2790: The operand of a 'delete' operator must be optional.`

It is wrong, because property `func` is defined in the instance prototype, not the instance itself and we set to the instance manually. Therefore we should be able to remove it to rollback to the prototype behavior

### 🙂 Expected behavior

No compiler error and the property is used again from the prototype as before

**Comment by fatcerberus** [comment]
The type system doesn't make a distinction between prototype properties and instance properties

**Comment by jcalz** [comment]
depends on #51073 --> #9726
